src/librustc_codegen_llvm/attributes.rs

```diff
@@ -164,6 +157,39 @@ pub fn from_fn_attrs(
 
     inline(cx, llfn, codegen_fn_attrs.inline);
 
+    match codegen_fn_attrs.optimize {
+        OptimizeAttr::None => {
+            match cx.tcx.sess.opts.optimize {
+                OptLevel::Size => {
+                    llvm::Attribute::MinSize.unapply_llfn(Function, llfn);
+                    llvm::Attribute::OptimizeForSize.apply_llfn(Function, llfn);
+                    llvm::Attribute::OptimizeNone.unapply_llfn(Function, llfn);
+                },
+                OptLevel::SizeMin => {
```

Comment:
hmm this match arm is identical to the `OptLevel::Size => ...` one above it? Is that a mistake in the implementation? Or should we fold them together? Or are you keeping them separate because you anticipate future changes?